### PR TITLE
Add auto assign IP adress for floating ip module

### PR DESCRIPTION
**Pull request by kostya13**

### CODE REVIEW

**Comment by idanmo** [comment]
Hi,
Was just wondering why there's a develop branch and not only a master branch as in all our other repositories?

**Comment by kostya13** [comment]
It's more convenient to me to develop new unstable functionality in a separate branch. 
When it becomes stable, we will transfer it to the main master.

File: tests/integration/test_network_plugin.py [hunk 1]
```diff
@@ -40,15 +40,30 @@ def setUp(self):
     def tearDown(self):
         super(NatRulesOperationsTestCase, self).tearDown()
 
-    def test_nat_rules_create_delete(self):
-        self.assertNotIn(self.public_ip, collectExternalIps(
-            self._get_gateway()))
+    def test_nat_rules_create_delete_with_explicit_ip(self):
+        self.ctx.node.properties['floatingip'].clear()
+        self.ctx.node.properties['floatingip'].update({'public_ip': self.public_ip,
+                                                       'gateway': 'M966854774-4471'})
```
Comment (achirko) [inline tests/integration/test_network_plugin.py]: Please use vcloud_integration_test_config.json (https://github.com/cloudify-cosmo/tosca-vcloud-plugin/blob/net-develop/examples/vcloud_integration_test_config.json.example) to pass things like public ip and gateway to the test.

File: tests/integration/test_network_plugin.py [hunk 2]
```diff
@@ -40,15 +39,33 @@ def setUp(self):
     def tearDown(self):
         super(NatRulesOperationsTestCase, self).tearDown()
 
-    def test_nat_rules_create_delete(self):
-        self.assertNotIn(self.public_ip, collectExternalIps(
-            self._get_gateway()))
+
+    def test_nat_rules_create_delete_with_explicit_ip(self):
+        self.ctx.node.properties['floatingip'].clear()
+        self.ctx.node.properties['floatingip'].update(IntegrationTestConfig().get()['floatingip'])
```
Comment (achirko) [inline tests/integration/test_network_plugin.py:42]: Please add example configuration in examples/vcloud_integration_test_config.json.example

File: tests/integration/test_network_plugin.py [hunk 3]
```diff
@@ -40,15 +39,33 @@ def setUp(self):
     def tearDown(self):
         super(NatRulesOperationsTestCase, self).tearDown()
 
-    def test_nat_rules_create_delete(self):
-        self.assertNotIn(self.public_ip, collectExternalIps(
-            self._get_gateway()))
+
+    def test_nat_rules_create_delete_with_explicit_ip(self):
+        self.ctx.node.properties['floatingip'].clear()
```
Comment (achirko) [inline tests/integration/test_network_plugin.py]: I think this line is unnecessary - you're recreating self.ctx before every test in setUp

File: tests/integration/test_network_plugin.py [hunk 4]
```diff
@@ -40,15 +39,33 @@ def setUp(self):
...
+        check_external = lambda: isExternalIpAssigned(public_ip, self._get_gateway())
+        self.assertFalse(check_external())
+        floatingip.connect_floatingip()
+        self.assertTrue(check_external())
+        self.assertRaises(cfy_exc.NonRecoverableError, floatingip.connect_floatingip)
+        floatingip.disconnect_floatingip()
+        self.assertFalse(check_external())
+
+    def test_nat_rules_create_delete_with_autoget_ip(self):
+        self.ctx.node.properties['floatingip'].clear()
```
Comment (achirko) [inline tests/integration/test_network_plugin.py]: I think this line is unnecessary - you're recreating self.ctx before every test in setUp

File: tests/integration/test_network_plugin.py [hunk 5]
```diff
@@ -16,18 +16,17 @@ class NatRulesOperationsTestCase(TestCase):
     def setUp(self):
         super(NatRulesOperationsTestCase, self).setUp()
 
+
```
Comment (achirko) [inline tests/integration/test_network_plugin.py]: Two blank lines in function body. Please use flake8 to unsure coding standards.

File: tests/integration/test_network_plugin.py [hunk 6]
```diff
@@ -40,15 +39,33 @@ def setUp(self):
...
+        self.assertTrue(check_external())
+        self.assertRaises(cfy_exc.NonRecoverableError, floatingip.connect_floatingip)
+        floatingip.disconnect_floatingip()
+        self.assertFalse(check_external())
+
+    def test_nat_rules_create_delete_with_autoget_ip(self):
+        self.ctx.node.properties['floatingip'].clear()
+        self.ctx.node.properties['floatingip'].update(IntegrationTestConfig().get()['floatingip'])
+        del self.ctx.node.properties['floatingip']['public_ip']
+        import pdb; pdb.set_trace()
```
Comment (achirko) [inline tests/integration/test_network_plugin.py]: Please delete it.

**Pull request merged**